packages/common/src/utils/stream-utils.ts

```diff
@@ -197,6 +196,10 @@ export class StreamUtils {
       }
     }
 
+    if (state.anchorStatus === AnchorStatus.ANCHORED) {
+      return true
+    }
+
     if (state.anchorStatus != base.anchorStatus) {
```

Comment:
```suggestion
    if (state.log.length == base.log.length && state.anchorStatus != base.anchorStatus) {
```

Reply:
So I think the thing this code was trying to deal with is two states that are otherwise identical, but where one has information about the anchor status and the other doesn't.  In that case you want to prefer the log with the anchor information.

The oversight with how it was written is that `state` could be **ahead** of `base` and legitimately have a different anchor status but still be a superset of `base`.  So I think if you make the change I propose here you can remove the `if(state.anchorStatus === AnchorStatus.ANCHORED)` check you added above.

Reply:
The suggestion makes sense. Thanks.